fecgov/openFEC
# refine H4 endpoint, add tests

## Summary 

- Resolves #5361 
- Resolves #4841 

 This ticket refines and deploys the H4 endpoint. I added tsvector filters for payee_name and disbursement_purpose, edited the other filters, added some automated testing, and removed the sort_expression logic. 

This should allow for @JonellaCulmer's design [here](https://app.mural.co/t/fec4461/m/fec4461/1680123342880/4b23fc49994eb22d3479d0407543d334ac42e044?sender=a8d29361-79af-49ec-a9b3-f5560b7ef3c5)


### Required reviewers

2 devs

## Impacted areas of the application

-  H4 endpoint

## Screenshots

![Screen Shot 2023-04-11 at 8 34 46 AM](https://user-images.githubusercontent.com/66386084/231163782-188202ba-fb4f-44cc-8f15-add0a54ca03b.png)

## How to test

Here's the[ data mapping PR](https://github.com/fecgov/openFEC/issues/5360) for reference and here's the [map doc](https://docs.google.com/document/d/1_0SqudoNoAXBdauhDkhalW3LIYbyqPo_HU6aY8VOg4Y/edit)

I have locust tested it and didn't see a large change. 

- `pyenv virtualenv 3.9.15 api15`
- `pyenv activate api15`
- `dropdb cfdm_test`
-  `createdb cfdm_test`
- `invoke create_sample_db`
-  http://127.0.0.1:5000/v1/schedules/schedule_h4/?q_disbursement_purpose=admin&sort_hide_null=false&sort_null_only=false&sort=event_purpose_date&per_page=20
- http://127.0.0.1:5000/v1/schedules/schedule_h4/?&sort_hide_null=false&sort_null_only=false&sort=event_purpose_date&per_page=20

It's currently on dev:
https://fec-dev-api.app.cloud.gov/v1/schedules/schedule_h4/?q_payee_name=Fed&sort=-event_purpose_date&sort_null_only=false&sort_hide_null=false&per_page=20
(28418 results)
